app/cafe.py

```diff
@@ -0,0 +1,20 @@
+from datetime import date
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if "vaccine" not in visitor:
+            raise NotVaccinatedError("You need to be vaccinated!")
+        elif visitor["vaccine"]["expiration_date"] < date.today():
+            raise OutdatedVaccineError("Your date of vaccine is expired!")
+        elif not visitor["wearing_a_mask"]:
+            raise NotWearingMaskError("You don't have mask!")
```

Comment:
Code Efficiency Issue: You are checking multiple conditions in a sequence using `if`, `elif` statements. This is not wrong, but it can be optimized. Consider using exceptions directly. Python has a philosophy of 'Easier to ask for forgiveness than permission'. This means that instead of checking if a condition is met before performing an action, you should perform the action and handle the exception if it occurs.

Reply:
agree

Reply:
just use `if` instead of `elif` as would be raised exactly one error, not more